### DISCUSSION TITLE
Help: Job finishes with no error but is never moved to completed

### DISCUSSION THREAD

**Discussion by bvkimball** [taskforcesh/bullmq#2965] [Q&A]
I have a logs all the way through my job, know it finishes successfully.
In my local dev environment it also works and moves to completed

In Staging environment it fails, things that are different:

- This job is my first job with a "returnvalue", all previous jobs still "complete" in Staging.  The returnvalue was a simple object but now it is just a string (both didn't fix the problem)
- I was on version 5.14, but updated to latest 5.34 to get deduplication logic
- Added deduplication logic:  ```{ deduplication: { id: `${projectId}-${user.role}`, ttl: FIVE_MINS } }```
- failedReason is: "job stalled more than allowable limit", but i think this is a by product of the real issue, because the job takes less than 3 seconds but never completes.

I have logs and try-catches everywhere, with nothing logging.

Any help or suggestion in debugging will be appreciated.

**Comment by bvkimball** [comment]
Update:
I am getting an error:  "Missing lock for job X. moveToFinished"

I have looked at everything in troubleshooting: https://docs.bullmq.io/guide/troubleshooting
But I am certain none of these are the issue, except the maxmemory since i can't set that with upstash.

I have try debugging to see if the lock is ever created... i don't ever see it while querying redis.  So my assumption is that the lock is being removed almost immediately or never set to begin with.

**Reply by manast** [reply]
One way this could happen is if your job is consuming all or almost all CPU and does not leave a chance to the worker to update the lock.

**Reply by manast** [reply]
Btw, by default the locks will live 30 seconds, but will be renewed every 15 seconds.

**Reply by manast** [reply]
Could you try with this new release? v5.34.2
